charts/athens-proxy/values.yaml

```diff
@@ -94,6 +94,16 @@ storage:
     # as GCP figures out internal authentication between products for you.
     serviceAccount: ""
 
+# Container security context configuration
+# This will override the `image.runAsNonRoot` settings in the specified container if `runAsUser` or `runAsGroup` are set
+securityContext: {}
+  # allowPrivilegeEscalation: false
+  # runAsNonRoot: true
+
+initContainerSecurityContext: {}
+  # allowPrivilegeEscalation: false
+  # runAsNonRoot: true
```

Comment:
How does this work? runAsNonRoot is no valid securityContext element, is it? This should state the following as an example, right?

```
       runAsUser: 1000
       runAsGroup: 1000
```

Reply:
Sorry, I didn't explain the change I'm making at all in this PR. 

There are two types of `securityContext`, one at the pod spec level (see [`PodSecurityContext` in the API reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#podsecuritycontext-v1-core) and one at the container spec level (see [`SecurityContext` in the API reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#securitycontext-v1-core)). My PR moves the existing `securityContext` set using the `image.runAsNonRoot` value to the `PodSecurityContext` spec level to ensure backwards compatibility and ensure this isn't a breaking change. By adding the additional `securityContext` and `initContainerSecurityContext` values in the chart it allows us to override the existing configuration with additional security context configuration options in each of the containers. 

I'll update the PR explanation and commit message to make the change clearer.

Reply:
I have updated the PR description to explain the change and added this to the commit message of the main commit. Hope this clears up what I'm trying to achieve with this change.

Reply:
Always learning something new, thanks for explaining!